src/map/battle.c

```diff
@@ -6735,7 +6735,9 @@ int battle_check_target( struct block_list *src, struct block_list *target,int f
 				&& md->guardian_data && (md->guardian_data->g || md->guardian_data->castle->guild_id) )
 				return 0; // Disable guardians/emperium owned by Guilds on non-woe times.
 
-			if (md->special_state.ai == AI_NONE) {
+			if (status_get_mode(s_bl) & MD_CANATTACK_MOB && t_bl->type == BL_MOB && status_get_mode(t_bl) & MD_CANATTACK_MOB)
```

Comment:
for performance reason better check t_bl->type == BL_MOB first.
for avoid mistakes ``a & b`` better always put into brackets like ``(a & b)``